version: '3.7'

services:
  django:
    restart: always
    container_name: dlp_django
    image: dlp_django
    build:
      context: .
      dockerfile: Dockerfile
    depends_on:
      db:
        condition: service_healthy
    env_file:
      - .env
    environment:
      - DJANGO_SETTINGS_MODULE=core.settings
    volumes:
      - .:/app
    ports:
      - '8000:8000'

  tasks_app:
    container_name: tasks_app
    image: dlp_django
    depends_on:
      - django
      - db
    env_file:
      - .env
    environment:
      - DJANGO_SETTINGS_MODULE=core.settings
    command: ['python', 'run.py']

  db:
    image: mysql:8.3.0
    user: mysql
    container_name: dlp_db
    restart: always
    volumes:
      - db_data:/var/lib/mysql
    env_file:
      - .env
    ports:
      - '3307:3306'
    healthcheck:
      test: ["CMD", "mysql", "-h", "localhost", "-u", "root", "-p${MYSQL_ROOT_PASSWORD}", "-e", "SELECT 1"]
      timeout: 5s
      retries: 10

volumes:
  db_data:
